### PR TITLE
fix(bot): improve jsonrpc client to support long inputs

### PR DESCRIPTION
**Pull request by Tommytrg**

### CODE REVIEW

File: packages/bot/src/nodeMethods/WitnetNodeClient.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+    this.client.on('close', () => {
+      if (callbackDone) {
+        callbackDone(result)
+      }
+    })
+    return result
+  }
+
+  async connect (port: number, host: string, cb: () => void) {
```
Comment (mariocao) [inline packages/bot/src/nodeMethods/WitnetNodeClient.ts:45]:
```suggestion
  async connect (host: string, port: number, cb: () => void) {
```

File: packages/bot/src/nodeMethods/WitnetNodeClient.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  private async callApiMethod (
+    methodName: string,
+    params: {} = {},
+    callbackDone?: (param?: any) => void
+  ): Promise<any> {
+    const request = {
+      jsonrpc: '2.0',
+      id: '1',
+      method: methodName,
+      params: `${params ? JSON.stringify(params) : JSON.stringify({})}}\n`
```
Comment (tmpolaczyk) [inline packages/bot/src/nodeMethods/WitnetNodeClient.ts:23]: I think this \n is a syntax error

File: packages/bot/src/nodeMethods/WitnetNodeClient.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    }
+    this.client.write(JSON.stringify(request))
```
Comment (tmpolaczyk) [inline packages/bot/src/nodeMethods/WitnetNodeClient.ts:25]: You also need to write a \n after the request to make sure that the node processes the request

File: packages/bot/src/nodeMethods/WitnetNodeClient.ts [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+    const result = await new Promise(resolve => {
+      let content = ''
+      this.client.on('data', chunk => {
+        content += chunk.toString()
+        if (chunk.toString().includes('\n')) {
```
Comment (tmpolaczyk) [inline packages/bot/src/nodeMethods/WitnetNodeClient.ts:31]: Technically every \n is one message, so if the data contains more than one \n we should parse multiple responses. But I guess we only support one single response per client?

Comment (Tommytrg) [reply packages/bot/src/nodeMethods/WitnetNodeClient.ts:31]: For the time being, we only need to support a single response per client

**Review by mariocao — APPROVED** [review]

File: packages/bot/src/nodeMethods/waitForTally.ts [hunk 5]
```diff
@@ -1,46 +1,28 @@
...
-  })
-  client.on('data', function (data: any) {
-    console.log('Received: ' + data)
-    let dataValue = JSON.parse(data)
-    if (dataValue && dataValue.result && dataValue.result.tally) {
-      let tally = dataValue.result.tally
-      let tallyBytes = JSON.stringify(tally.tally)
-      console.log('Found tally for DR ' + drTxHash + ': ' + tallyBytes)
-      callbackFound(tally)
-      client.destroy() // kill client after server's response
```
Comment (tmpolaczyk) [inline packages/bot/src/nodeMethods/waitForTally.ts:31]: Same, this line is missing.

**Pull request merged**